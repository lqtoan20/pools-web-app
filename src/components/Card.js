import { connect } from "react-redux";
import { Link } from "react-router-dom";

const Card = ({ question, author }) => {
  return (
    <Link to={"questions/" + question.id}>
      <div className="uk-card uk-card-default uk-card-hover">
        <div className="uk-card-header">
          <div className="uk-grid-small uk-flex-middle uk-grid">
            <div className="uk-width-auto">
              <img
                className="uk-border-circle"
                width="40"
                height="40"
                src={author?.avatarURL}
                alt="Avatar"
              />
            </div>
            <div className="uk-width-expand">
              <h3 className="uk-card-title uk-margin-remove-bottom">
                {question.author}
              </h3>
              <p className="uk-text-meta uk-margin-remove-top">
                {" "}
                {new Date(question.timestamp).toDateString()}
              </p>
            </div>
          </div>
        </div>
        <div className="uk-card-footer">
          <button className="uk-button uk-button-text">Show</button>
        </div>
      </div>
    </Link>
  );
};

export default connect()(Card);
