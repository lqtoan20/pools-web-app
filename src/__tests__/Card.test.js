import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux"; // For providing the store
import Card from "../components/Card"; // Adjust the path accordingly
import { store } from "../stores/store";
import { BrowserRouter } from "react-router-dom";

// Basic test to check if the component renders correctly
test("Card component renders correctly", () => {
  const question = {
    id: "123",
    author: "sarahedo",
    timestamp: 1630070400000,
  };
  const author = {
    id: "sarahedo",
    avatarURL: "https://example.com/avatar.jpg",
  };

  const { getByText } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Card question={question} author={author} />
      </BrowserRouter>
    </Provider>
  );

  expect(getByText("sarahedo")).toBeInTheDocument();
  expect(getByText("Fri Aug 27 2021")).toBeInTheDocument();
});

// Snapshot test to check if the component's UI matches the snapshot
test("Card component matches snapshot", () => {
  const question = {
    id: "123",
    author: "sarahedo",
    timestamp: 1630070400000,
  };
  const author = {
    id: "sarahedo",
    avatarURL: "https://example.com/avatar.jpg",
  };

  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <Card question={question} author={author} />
      </BrowserRouter>
    </Provider>
  );

  expect(component).toMatchSnapshot();
});
