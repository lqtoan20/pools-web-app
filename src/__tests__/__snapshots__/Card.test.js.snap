// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/questions/123"
      >
        <div
          class="uk-card uk-card-default uk-card-hover"
        >
          <div
            class="uk-card-header"
          >
            <div
              class="uk-grid-small uk-flex-middle uk-grid"
            >
              <div
                class="uk-width-auto"
              >
                <img
                  alt="Avatar"
                  class="uk-border-circle"
                  height="40"
                  src="https://example.com/avatar.jpg"
                  width="40"
                />
              </div>
              <div
                class="uk-width-expand"
              >
                <h3
                  class="uk-card-title uk-margin-remove-bottom"
                >
                  sarahedo
                </h3>
                <p
                  class="uk-text-meta uk-margin-remove-top"
                >
                   
                  Fri Aug 27 2021
                </p>
              </div>
            </div>
          </div>
          <div
            class="uk-card-footer"
          >
            <button
              class="uk-button uk-button-text"
            >
              Show
            </button>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/questions/123"
    >
      <div
        class="uk-card uk-card-default uk-card-hover"
      >
        <div
          class="uk-card-header"
        >
          <div
            class="uk-grid-small uk-flex-middle uk-grid"
          >
            <div
              class="uk-width-auto"
            >
              <img
                alt="Avatar"
                class="uk-border-circle"
                height="40"
                src="https://example.com/avatar.jpg"
                width="40"
              />
            </div>
            <div
              class="uk-width-expand"
            >
              <h3
                class="uk-card-title uk-margin-remove-bottom"
              >
                sarahedo
              </h3>
              <p
                class="uk-text-meta uk-margin-remove-top"
              >
                 
                Fri Aug 27 2021
              </p>
            </div>
          </div>
        </div>
        <div
          class="uk-card-footer"
        >
          <button
            class="uk-button uk-button-text"
          >
            Show
          </button>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
